### DISCUSSION TITLE
Sorting doesn't work on nova resource but works on relations

### DISCUSSION THREAD

**Discussion by anstapol** [outl1ne/nova-sortable#156] [Q&A] [answered]
I've went through the configuration guide the sorting on relations works but `HasSortableRows` trait has no effect on the nova resource.

laravel/framework – v9.25.1
laravel/nova – 4.12.14

spatie/eloquent-sortable – 4.0.1
outl1ne/nova-sortable - 3.2.1

**Comment by KasparRosin** [comment, marked as answer]
Hey!
Try out the latest version to see whether the issue remains.
Unable to reproduce this with a default configuration.

**Reply by anstapol** [reply]
Yeah I think it got fixed on some release, sorry totally forgot to remove this 😅

Thank you for your answer!

**Comment by Israel5** [comment]
I'm having the same issue.
I have had this issue for a long time and seeing the last interaction, I decided to update all packages to the latest version, but it didn't solve the problem.

I'm able to sort the relations, as can be seen below:
<img width="354" alt="image" src="https://user-images.githubusercontent.com/63180786/213081917-526fe7b1-45cb-4475-87da-a382c9c92ada.png">

But I can't make it work on the resource directly:
<img width="309" alt="image" src="https://user-images.githubusercontent.com/63180786/213082955-2bd88237-b1ca-4378-aab0-7599d5ca821f.png">

I'm using the following package versions:
Laravel Framework: `9.48.0`
Laravel Nova: `4.21.0`
outl1ne/nova-sortable: `3.4.2`
spatie/eloquent-sortable: `4.0.1`

It's a private repo of a client so I can't share it here, but I follow the same guides available. The only difference is that instead of `'order_column_name' => 'sort_order'`, I'm using `'order_column_name' => 'sort'`.

---

Basically this is what I have in the code:

`app/Models/Phase.php`
```php
<?php

namespace App\Models;

use App\Traits\Uuid;
use App\Traits\Observable;
use Spatie\MediaLibrary\HasMedia;
use Spatie\EloquentSortable\Sortable;
use Illuminate\Database\Eloquent\Model;
use Spatie\EloquentSortable\SortableTrait;
use Spatie\MediaLibrary\InteractsWithMedia;
use Illuminate\Database\Eloquent\Factories\HasFactory;

class Phase extends Model implements Sortable, HasMedia
{
    use HasFactory;
    use Uuid;
    use Observable;
    use SortableTrait;
    use InteractsWithMedia;

   ...

   /**
      * The attributes that should be sortable.
      *
      * @var array<string, string>
      */
     public $sortable = [
         'order_column_name' => 'sort',
         'sort_when_creating' => true,
         'sort_on_has_many' => true,
     ];
```

`app/Nova/Product.php`
```php
<?php

namespace App\Nova;

use Laravel\Nova\Fields\Text;
use Laravel\Nova\Fields\Boolean;
use Laravel\Nova\Fields\HasMany;
use Laravel\Nova\Fields\Textarea;
use Laravel\Nova\Http\Requests\NovaRequest;
use App\Actions\Uploads\RenameUploadFilesAction;
use Outl1ne\NovaSortable\Traits\HasSortableRows;
use Ebess\AdvancedNovaMediaLibrary\Fields\Files;

class Phase extends Resource
{
    use HasSortableRows;
   
   ...
```

**Reply by KasparRosin** [reply]
Sorry. For some reason I thought the main post was talking about ManyToMany sorting.

HasMany relationship does not support sorting on both relation and index views.
https://github.com/outl1ne/nova-sortable#sorting-on-manytomany-relationships
> NB! The resource can only be sorted on either the Index view or the HasMany list view, but not both!

**Comment by Israel5** [comment]
Hi @KasparRosin, thank you for your answer.

I believe I might be missing some steps here, but just to give you another concrete example.
I'm using your package in another Laravel application for a customer, and I'm basically doing the same thing I described above.

<img width="238" alt="image" src="https://user-images.githubusercontent.com/63180786/214145860-b6a7c027-d51d-47f6-856e-60d5fd17762a.png">

So, the code running Laravel Nova 3.22 works fine. Whenever I enter the resource, I'm able to see the icons to drag the records to the position I want. (see it above)

But, the application running Laravel Nova 4.21, which has the same implementation, does not show the icons to sort the records when I  enter the resource, only if I load it as a relationship.
